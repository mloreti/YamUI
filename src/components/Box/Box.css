/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
@import "src/css/variables/colors.css";

.y-box {
  position: relative;
  border-radius: 1px;
  border: 1px solid $borderColor;
}

.y-box--inner {
  margin: -1px;
}
