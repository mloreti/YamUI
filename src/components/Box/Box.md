### Examples

Basic usage:

```js { "props": { "data-description": "basic" } }
<Box>This is a Box</Box>
```

With onClick:

```js { "props": { "data-description": "with onClick" } }
<Box onClick={action('onClick')}>Click me</Box>
```
