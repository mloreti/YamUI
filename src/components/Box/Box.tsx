/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { NestableBaseComponentProps as BoxProps } from '../../util/BaseComponent/props';
import { join } from '../../util/classNames';
import Block, { GutterSize } from '../Block';
import './Box.css';

export { BoxProps };
/**
 * A Box is simply a div with a border and fixed padding. It also has `display: relative` so you can
 * absolutely-position elements like a close button. Boxes will primarily be used for card views.
 */
export default class Box extends React.Component<BoxProps, {}> {
  render() {
    const { children, className } = this.props;

    return (
      <Block className={join(['y-box', className])} padding={GutterSize.SMALL}>
        <div className="y-box--inner">{children}</div>
      </Block>
    );
  }
}
