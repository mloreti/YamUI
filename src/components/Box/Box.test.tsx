/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Box, { BoxProps } from './index';

describe('<Box />', () => {
  let component: ShallowWrapper<BoxProps, {}>;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<Box />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<Box className="TEST_CLASSNAME" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with child content', () => {
    beforeEach(() => {
      component = shallow(<Box>child content</Box>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
