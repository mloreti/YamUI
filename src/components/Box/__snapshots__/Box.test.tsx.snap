// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> with additional className matches its snapshot 1`] = `
<Block
  className="y-box TEST_CLASSNAME"
  padding="small"
>
  <div
    className="y-box--inner"
  />
</Block>
`;

exports[`<Box /> with child content matches its snapshot 1`] = `
<Block
  className="y-box"
  padding="small"
>
  <div
    className="y-box--inner"
  >
    child content
  </div>
</Block>
`;

exports[`<Box /> with default options matches its snapshot 1`] = `
<Block
  className="y-box"
  padding="small"
>
  <div
    className="y-box--inner"
  />
</Block>
`;
